core/src/main/groovy/com/novoda/gradle/release/AndroidArtifacts.groovy

```diff
@@ -42,7 +42,8 @@ class AndroidArtifacts implements Artifacts {
     }
 
     def mainJar(Project project) {
-        "$project.buildDir/outputs/aar/${project.name}-${variant.baseName}.aar"
+        def artifactBaseName = project.getProperty("archivesBaseName")
```

Comment:
what about checking if `archivesBaseName ` is defined, if so use it, if not fall back to `${project.name}-${variant.baseName}`?

Reply:
Sure it can be added as a fallback, but I would prefer not to. This seems like the only right way to determine the archive's name.

Do you agree, or would you like me to add the fallback anyway?